packages/drivers/odsp-driver/src/tokenFetch.ts

```diff
@@ -24,54 +24,70 @@ export interface TokenFetchOptions {
      */
     refresh: boolean;
 
-    /** Claims that have to be passed with token fetch request */
+    /**
+     * Claims that have to be passed with token fetch request.
+     * These can be used to specify additional information that must be passed to token authority.
+     */
     claims?: string;
+
+    /**
+     * Tenant id of authority that must be handling token fetch.
+     * If it is not specified then it is up to token fetching logic to determine which tenant authority
+     * to use to issue access token.
+     */
+    tenantId?: string;
 }
 
 /**
- * Method signature for callback method used to fetch Storage token
- * @param siteUrl - Storage site url
- * @param refresh - Flag indicating whether token fetch must bypass local cache
- * @param claims - Optional string indicating claims that will be passed to token authority
+ * Method signature for callback method used to fetch access token
+ * @param options - token fetch options
  * @returns If successful, TokenResponse object representing token value along with flag indicating
  * whether token came from cache. Legacy implementation may return a string for token value;
  * in this case it should be assumes that fromCache signal is undefined. Null is returned in case of failure.
  */
-// eslint-disable-next-line max-len
-export type StorageTokenFetcher = (siteUrl: string, refresh: boolean, claims?: string) => Promise<string | TokenResponse | null>;
+export type TokenFetcher = (options: TokenFetchOptions) => Promise<string | TokenResponse | null>;
+
+/**
+ * Represents access token fetch options for ODSP resource
+ */
+export interface OdspResourceTokenFetchOptions extends TokenFetchOptions {
+    /** Site url representing ODSP resource location */
+    siteUrl: string;
+}
 
 /**
- * Method signature for callback method used to fetch Push token
- * @param refresh - Flag indicating whether token fetch must bypass local cache
- * @param claims - Optional string indicating claims that will be passed to token authority
+ * Method signature for callback method used to fetch token for accessing ODSP resource with location
+ * represented by site url.
+ * @param options - token fetch options
  * @returns If successful, TokenResponse object representing token value along with flag indicating
  * whether token came from cache. Legacy implementation may return a string for token value;
  * in this case it should be assumes that fromCache signal is undefined. Null is returned in case of failure.
  */
-export type PushTokenFetcher = (refresh: boolean, claims?: string) => Promise<string | TokenResponse | null>;
+export type OdspResourceTokenFetcher =
```

Comment:
Idea: define a single `TokenFetcher<T>` type where `T` extends `TokenFetchOptions` instead of indivdual `FooFetcher` types?

Reply:
Good idea